### ISSUE TITLE
(timestream): retentionProperties as Json is not working anymore

### ISSUE THREAD

**Issue by Wurstnase** [aws/aws-cdk#23404]
### Describe the bug

With 2.55.0 the cfnspec changes.

https://github.com/aws/aws-cdk/commit/3951f09fb370b74936f3fb45e7188ac6c7343b67#diff-0d394a72825e270916f40b00a99f6e46b17ad80d4a4c76d19dd1688883a0f96cR451
https://github.com/aws/aws-cdk/commit/3951f09fb370b74936f3fb45e7188ac6c7343b67#diff-0d394a72825e270916f40b00a99f6e46b17ad80d4a4c76d19dd1688883a0f96cR457
```diff
      "MagneticStoreWriteProperties": {
          "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-timestream-table.html#cfn-timestream-table-magneticstorewriteproperties",
-         "PrimitiveType": "Json",
          "Required": false,
+         "Type": "MagneticStoreWriteProperties",
          "UpdateType": "Mutable"
        },
        "RetentionProperties": {
          "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-timestream-table.html#cfn-timestream-table-retentionproperties",
-          "PrimitiveType": "Json",
          "Required": false,
+          "Type": "RetentionProperties",
          "UpdateType": "Mutable"
        },
```
At least with python, when you define the structure as a python-`dict`, the properties are missing in the cloudformation template.

```diff
        timestream.CfnTable(
            self,
            "Table",
            database_name=cfn_database.ref,
            table_name=TABLE_NAME,
-            retention_properties={
-                "MagneticStoreRetentionPeriodInDays": magnetic_store_retention_period_in_days,
-                "MemoryStoreRetentionPeriodInHours": memory_store_retention_period_in_hours,
+            retention_properties=timestream.CfnTable.RetentionPropertiesProperty(
+                magnetic_store_retention_period_in_days=str(magnetic_store_retention_period_in_days),
+                memory_store_retention_period_in_hours=str(memory_store_retention_period_in_hours),
            ),
        )
```

### Expected Behavior

Defining the properties as a python-dict should synth the properties into the cloudformation stack.

### Current Behavior

When defining the properties as a dict, the cloudformation stack has not properties.

### Reproduction Steps

see python example above.

### CDK CLI Version

2.55.1

### Node.js Version

-

### OS

-

### Language

Python

### Language Version

3.9.16

**Comment by peterwoodworth** [comment]
I'm sorry this regression occurred @Wurstnase, we try to avoid merging spec changes which will cause regressions for customers. Thank you for letting us know this happened 🙂

**Comment by Wurstnase** [comment]
Thanks for your quick response. Fortunately, we have snapshot tests for our stacks.

The most annoying thing about this error is that it is silent. Not an error. It just disappears. Maybe there is an idea that this could lead to an exception?

**Comment by madeline-k** [comment]
Hey @Wurstnase, this has been fixed and released in version 2.56.0! We are keeping track of occurrences like this in this issue: https://github.com/aws/aws-cdk/issues/21767, and solutions/suggestions for this problem should go there. I like your suggestion of having an error message. It might be tricky to do since this manifests slightly differently in each target language.
